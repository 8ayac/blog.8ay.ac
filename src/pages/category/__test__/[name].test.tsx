import CategoryPage from '@/src/pages/category/[name]';
import { mount, shallow } from 'enzyme';
import React from 'react';

describe('CategoryPage', () => {
  const testCategoryName = 'test-cat';
  const testArticles = [
    {
      id: 'test-article-01',
      title: 'Test Title 01',
      tags: [testCategoryName],
      publishedAt: '2021-01-01 00:00:00 +0000',
      updatedAt: '2021-01-01 00:00:00 +0000',
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-02',
      title: 'Test Title 02',
      tags: [testCategoryName],
      publishedAt: '2021-02-02 00:00:00 +0000',
      updatedAt: '2021-02-02 00:00:00 +0000',
      body: 'bluhbluhbluh',
    },
    {
      id: 'test-article-03',
      title: 'Test Title 03',
      tags: [testCategoryName],
      publishedAt: '2021-03-03 00:00:00 +0000',
      updatedAt: '2021-03-03 00:00:00 +0000',
      body: 'bluhbluhbluh',
    },
  ];

  describe('is rendered correctly', () => {
    test('to match the snapshot', () => {
      const wrapper = shallow(
        <CategoryPage name={testCategoryName} filteredArticles={testArticles} />
      );
      expect(wrapper).toMatchSnapshot();
    });

    test('title header of the page', () => {
      const wrapper = mount(
        <CategoryPage name={testCategoryName} filteredArticles={testArticles} />
      );
      expect(wrapper.find('BodyHeader').first().text()).toBe(
        `Tag: ${testCategoryName}`
      );
    });
  });

  describe('has proper style rules', () => {
    const wrapper = shallow(
      <CategoryPage name={testCategoryName} filteredArticles={testArticles} />
    );

    test('in ArticleListWrapperDiv', () => {
      expect(wrapper.find('ArticleListWrapperDiv')).toHaveStyleRule(
        'max-width',
        '90%'
      );
      expect(wrapper.find('ArticleListWrapperDiv')).toHaveStyleRule(
        'margin',
        '0 auto'
      );
    });
  });
});
