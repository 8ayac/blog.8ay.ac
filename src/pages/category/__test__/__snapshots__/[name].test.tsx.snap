// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoryPage is rendered correctly to match the snapshot 1`] = `
<Fragment>
  <BodyHeader
    title="Tag: test-cat"
  />
  <ArticleListWrapperDiv>
    <ArticleHeaderList
      articles={
        Array [
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-01",
            "publishedAt": "2021-01-01 00:00:00 +0000",
            "tags": Array [
              "test-cat",
            ],
            "title": "Test Title 01",
            "updatedAt": "2021-01-01 00:00:00 +0000",
          },
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-02",
            "publishedAt": "2021-02-02 00:00:00 +0000",
            "tags": Array [
              "test-cat",
            ],
            "title": "Test Title 02",
            "updatedAt": "2021-02-02 00:00:00 +0000",
          },
          Object {
            "body": "bluhbluhbluh",
            "id": "test-article-03",
            "publishedAt": "2021-03-03 00:00:00 +0000",
            "tags": Array [
              "test-cat",
            ],
            "title": "Test Title 03",
            "updatedAt": "2021-03-03 00:00:00 +0000",
          },
        ]
      }
    />
  </ArticleListWrapperDiv>
</Fragment>
`;
